src/platform/plugins/shared/discover/public/application/main/components/single_tab_view/main_app.tsx

```diff
@@ -41,13 +41,6 @@ export function DiscoverMainApp({ stateContainer }: DiscoverMainProps) {
     addHelpMenuToAppChrome(chrome, docLinks);
   }, [chrome, docLinks]);
 
-  useEffect(() => {
-    return () => {
-      // clear session when navigating away from discover main
-      data.search.session.clear();
-    };
-  }, [data.search.session]);
-
```

Comment:
I'm wondering if this will have some unintended side effects - possibly with canceling in-progress requests or including search requests from this session in a new session if navigating from one app to another... Just flagging this as something to check, I haven't identified real issues yet

Reply:
Just noticed it's moved, not removed entirely 😄